### PR TITLE
exit server if not all Ark CRDs exist at startup

### PR DESCRIPTION
**Pull request by skriss**
Signed-off-by: Steve Kriss <steve@heptio.com>

### CODE REVIEW

File: pkg/apis/ark/v1/register.go [hunk 1]
```diff
@@ -41,27 +41,32 @@ func Resource(resource string) schema.GroupResource {
...
 }
 
+type typeInfo struct {
+	ItemType     runtime.Object
+	ItemListType runtime.Object
+}
+
+// CustomResources returns a map of all custom resources within the Ark
+// API group, keyed on Kind.
+func CustomResources() map[string]typeInfo {
```
Comment (nrb) [inline pkg/apis/ark/v1/register.go]: Would it make sense to re-use https://github.com/heptio/ark/blob/master/pkg/install/crd.go here? That way we'd have them defined in one place.

Comment (skriss) [reply pkg/apis/ark/v1/register.go]: I'd rather go the other way around (have `pkg/install` use the `pkg/apis/ark/v1` func), but yeah good point, I'll update the install package.

Comment (nrb) [reply pkg/apis/ark/v1/register.go]: Yeah, that direction does make more sense. If we use one common source then we're less likely to miss requiring any new CRDs getting added.

Comment (skriss) [reply pkg/apis/ark/v1/register.go]: updated.

**Comment by skriss** [comment]
Working on adding a unit test for the server func.

**Comment by skriss** [comment]
added unit.

**Comment by skriss** [comment]
Manual testing looks good on this - if any CRD is missing, the server exits.  If all are present, the server proceeds.

**Comment by carlisia** [comment]
I 👀 👍

**Pull request merged**